bin/agoric

```diff
@@ -3,18 +3,22 @@
 esmRequire = require('esm')(module);
 const path = esmRequire('path');
 const chalk = esmRequire('chalk').default;
+const WebSocket = esmRequire('ws');
+const { spawn } = esmRequire('child_process');
 
 const main = esmRequire('../lib/main.js').default;
 const progname = path.basename(process.argv[1]);
 
 const error = (...args) => {
   console.error(`${progname}: ${chalk.red('ERROR')}:`, ...args);
 };
+const fs = require('fs').promises;
+const makeWebSocket = (...args) => new WebSocket(...args);
 
 process.on('SIGINT', () => process.exit(-1));
 
 const rawArgs = process.argv.splice(2);
-main(progname, rawArgs, { console, error }).then(
+main(progname, rawArgs, { console, error, makeWebSocket, fs }).then(
```

Comment:
Need to pass through `spawn` as well.

Reply:
Consider passing through `chalk`.  It's powerful in that it looks up the current terminal attributes.

Reply:
`bundleSource` from `@agoric/bundle-source` is also powerful, FWIW.

Reply:
> Need to pass through `spawn` as well.

I'm spoiled by static type checkers.

And it passes all the tests as is :)

Out of curiosity: how did you find this problem? Just staring at it? Or did you run a sort of integration test?

Reply:
Staring. 😛

Reply:
re chalk... isn't there some way to pass in access to the tty and such? ugh... doesn't look like it.

Reply:
`bundleSource` ... ugh! I'm inclined to open an issue there to support passing in path, rollup, and the node resolve plugin.

Reply:
I added the missing `spawn` (after rebasing to the monorepo).

For bundleSource, see https://github.com/Agoric/agoric-sdk/pull/371

I'll leave chalk for another day.